tests/test_models.py

```diff
@@ -0,0 +1,237 @@
+        parent.refresh_from_db()
+        assert status.state == UserTaskStatus.RETRYING
+        assert status.attempts == 2
+        assert parent.state == UserTaskStatus.IN_PROGRESS
+
+    def test_set_name(self):
+        """set_name() should update both the subject and any parent."""
+        parent = self._status(is_container=True)
+        status = self._status(parent=parent)
+        status.set_name('New Name')
+        parent.refresh_from_db()
+        assert status.name == 'New Name'
+        assert parent.name == 'New Name'
+
+    def test_set_state(self):
+        """It should be possible to set a custom state for a UserTaskStatus."""
+        status = self._status()
+        status.set_state('Custom State')
+        assert status.state == 'Custom State'
+        status.refresh_from_db()
+        assert status.state == 'Custom State'
+
+    def test_set_state_canceled(self):
+        """Setting a custom state for a canceled task should fail."""
+        status = self._status(state=UserTaskStatus.CANCELED)
+        with pytest.raises(TaskCanceledException):
+            status.set_state('Custom State')
+        assert status.state == UserTaskStatus.CANCELED
+        status.refresh_from_db()
+        assert status.state == UserTaskStatus.CANCELED
+
+    def test_set_state_parent(self):
+        """Setting a custom state should propagate to the state's parent."""
+        parent = self._status(is_container=True, task_class='celery.chain')
+        status = self._status(parent=parent)
+        status.set_state('Custom State')
+        assert status.state == 'Custom State'
+        status.refresh_from_db()
+        assert status.state == 'Custom State'
+        assert parent.state == 'Custom State'
+        parent.refresh_from_db()
+        assert parent.state == 'Custom State'
+
+    def test_start(self):
+        """The start() method should update the status record appropriately."""
+        status = self._status(state=UserTaskStatus.PENDING)
+        assert status.state == UserTaskStatus.PENDING
+        status.start()
+        assert status.state == UserTaskStatus.IN_PROGRESS
+        status.refresh_from_db()
+        assert status.state == UserTaskStatus.IN_PROGRESS
+
+    def test_start_canceled_task(self):
+        """Attempting to start a canceled task should raise a TaskCanceledException."""
+        status = self._status(state=UserTaskStatus.CANCELED)
+        with pytest.raises(TaskCanceledException):
+            status.start()
+        assert status.state == UserTaskStatus.CANCELED
+
+    def test_start_with_failed_sibling(self):
+        """If a sibling task has already failed, the parent status should not change when the task starts."""
+        parent = self._status(is_container=True, state=UserTaskStatus.FAILED)
+        child1 = self._status(parent=parent, state=UserTaskStatus.FAILED)
+        child2 = self._status(parent=parent, state=UserTaskStatus.PENDING)
+        child2.start()
+        parent.refresh_from_db
```

Comment:
Forgot to add linting of the tests directory, which should have caught that.  Adding that both here and in the cookiecutter.